BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -154,8 +154,79 @@ def _install_action(ctx, infos, itarget):
     )
     return [output]
 
+SourceOutputFileMapInfo = provider(
+    doc = "...",
+    fields = {
+        "mapping": "Dictionary where keys are source file paths and values are at the respective .o file under bazel-out",
+    },
+)
+
+def _source_output_file_map(target, ctx):
+    """
+    Maps source code files to respective `.o` object file under bazel-out. Output group is used for indexing in xcbuildkit.
+    """
+    source_output_file_map = ctx.actions.declare_file("{}_source_output_file_map.json".format(target.label.name))
+
+    mapping = {}
+    objc_srcs = []
+    objc_objects = []
+
+    # List of source files to be mapped to output files
+    if hasattr(ctx.rule.attr, "srcs"):
+        objc_srcs = [
+            f
+            for source_file in ctx.rule.attr.srcs
+            for f in source_file.files.to_list()
+            # Handling objc only for now
+            if f.path.endswith(".m")
```

Comment:
Missing other lang types ? `.c` `.cpp` `.mm` `.cxx` ?

Reply:
Good catch, will add the other types here using `rules_ios` as a reference